src/main/webapp/assets/script.js

```diff
@@ -222,16 +225,92 @@ function initMap() {
 
   // Get all posts on the page and show them as markers.
   posts.forEach((post) => {
+    const width = getMapMarkerIconSize(post.numOfPeopleFoodWillFeed, 'width');
+    const height = getMapMarkerIconSize(post.numOfPeopleFoodWillFeed, 'height');
+    const icon = {
+      url: getMapMarkerIconUrl(post.eventStartTime),
+      scaledSize: new google.maps.Size(width, height),
+      origin: new google.maps.Point(0, 0),
+      anchor: new google.maps.Point(0, 0),
+    };
+
     new google.maps.Marker({
       position: {lat: post.location.lat, lng: post.location.long},
       map: map,
       title: post.organizationName,
+      opacity: getMapMarkerOpacity(post.eventStartTime, post.eventEndTime),
+      icon: icon,
     });
   });
 
 /* eslint-enable no-undef */
 }
 
+/**
+ * Calculates the opacity of a given map marker.
+ * @param {Date} eventStartTime - The start time of the event.
+ * @param {Date} eventEndTime - The end time of the event.
+ * @return {number} - The opacity to show the marker as.
+ */
+function getMapMarkerOpacity(eventStartTime, eventEndTime) {
+  const now = new Date();
+  const startTime = eventStartTime;
+  const endTime = eventEndTime;
+
+  // If the date is in the future, our marker should have full opacity.
+  if (startTime > now) {
+    return 1;
+  }
+
+  // If the event is over, our marker should not be on the map at all.
+  if (endTime < now) {
+    return 0;
+  }
```

Comment:
Why not have this is an `if () {} else if () {}` rather than two separate `if` statements since they're the only 2 edge cases anyways?

Reply:
In practice, both structures behave the same: if `startTime > now` evaluates to `true`, then the return statement will cause the second `if` to not evaluate, same as an `if ... else if` chain. I can use the proposed version, though

Reply:
Yes, they do have the same behavior so I was just wondering why you picked this version over the other! The two separate statements imply that the functionalities are two separate entities and should be tested separately which is not the case so I believe the proposed version would work better!

Reply:
Ah okay, no worries! The reason I had structured the way is honestly pretty inconsequential -- for me, stuff like this reads better when each case gets its own visual block. ESLint puts the `else if` on the same line, and so the whole thing becomes one block of text. I've added the proposed version in my newest commits though, your reasoning makes sense! :)

Reply:
Since this is a minor point and the style guide does not enforce this, I would say either is completely fine. I do not think Melanie's approach means that they are two different entities that need to be tested separated. Regardless of how it is done, it will require at least two tests for each part. I think for matters like this (no formal specification, logically the same, minimal readability improvements, working on limited time), it is fine to leave as is.